extensions/wikia/AdMixExperiment/scripts/noVideo1.js

```diff
@@ -0,0 +1,269 @@
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+			}
+
+			$adAndRecircWrapper.css({
+				bottom: '',
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function reset() {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+			resetRecircStyles();
+		}
+
+		function scrollAfterAdAndRecircWrapperTopPosition() {
+			var adAndRecircWrapperStyles = {
+				position: '',
+				width: ''
+			};
+
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', gapSize + 'px');
+
+				$recirc.css({
+					position: 'fixed',
+					top: globalNavigationHeight + 'px',
+					width: '280px'
+				});
+
+				adAndRecircWrapperStyles.top = '';
+			} else {
+				adAndRecircWrapperStyles.bottom = '';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAfterAdAndRecircWrapperBottomPosition() {
+			var adAndRecircWrapperStyles = {
+				position: 'fixed',
+				width: '300px'
+			};
+
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', '0');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+
+				adAndRecircWrapperStyles.top = globalNavigationHeight + 'px';
+			} else {
+				adAndRecircWrapperStyles.bottom = noRecircAdBottomPosition + 'px';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAtBottom(stopPosition) {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			// FIXME cache it? at least the element
+			var topAdHeight = $('#WikiaTopAds').outerHeight(true);
+			var adAndRecircWrapperStyles = {
+				position: 'absolute',
+				top: (stopPosition - topAdHeight) + 'px',
+				width: '300px'
+			};
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', '0');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+
+				adAndRecircWrapperStyles.top = (stopPosition - topAdHeight) + 'px';
+			} else {
+				adAndRecircWrapperStyles.bottom = noRecircAdBottomPosition + 'px';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function resetTopAd() {
+			$topRightAd.css({
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function resetTopAdPaddings() {
+			$topRightAdWrapper.css({
+				'padding-top': '',
+				'padding-bottom': gapSize + 'px'
+			});
+		}
+
+		function scrollAfterFirstAdTopPosition() {
+			resetRecircStyles();
+			resetTopAdPaddings();
+			$topRightAd.css({
+				position: 'fixed',
+				top: globalNavigationHeight + 'px',
+				width: '300px'
+			});
+		}
+
+		function scrollAfterFirstAdBottomPosition() {
+			resetRecircStyles();
+			resetTopAd();
+			$topRightAdWrapper.css({
+				'padding-bottom': '',
+				'padding-top': gapSize + 'px'
+			});
+		}
+
+		function update() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 14 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)